Update dependencies

nmajor25 opened the pull request:
Shortcut Story ID: [sc-40888]

nmajor25 committed:
Update dependencies

nmajor25 commented on package.json:
```diff
@@ -178,7 +178,7 @@
     "css-loader": "^6.2.0",
     "cssnano": "^6.0.0",
-    "cypress": "^12.14.0",
+    "cypress": "^13.2.0",
```
I think Cypress v13 is causing tests to fail.

But only on CircleCI. Everything looks good when I run them locally. So I'm not sure what's going on.

On the test replays it looks like the CSS isn't being loaded properly:

![Screenshot 2023-09-15 at 2 29 04 PM](https://github.com/RoundingWell/care-ops-frontend/assets/35355575/cad57327-6b96-4b8d-9dc2-1ad296c47b9f)

paulfalgout replied:
So what's cool about v13 is that we should be able to inspect the source maybe on cloud.cypress.io.. I'm not entirely sure though how in depth it gets or if it is still just essentially screenshots..

nmajor25 commented:
@paulfalgout there's two more outdated packages we could update:

1. `@cypress/webpack-preprocessor`: `v5.17.1` => `v6.0.0`
    * Nothing new, they accidentally created a release: https://github.com/cypress-io/cypress/blob/develop/npm/webpack-preprocessor/CHANGELOG.md

2. `stylelint-config-standard-scss`: `10.0.0` => `11.0.0`
    * Nothing too significant: https://github.com/stylelint-scss/stylelint-config-standard-scss/blob/main/CHANGELOG.md#1100

paulfalgout commented:
Yeah we should do all the updates.. if we have to wait on cypress we can pull that one back.  I did fill out a report with them

nmajor25 commented:
@paulfalgout okay I've got Cypress switched back to use the newest iteration of v12. And `@cypress/webpack-preprocessor` and `stylelint-config-standard-scss` have been updated.

The `npm outdated` command will now show only `formio` and `cypress` as needing to be updated.

nmajor25 committed:
Update CircleCI to use Node.js v18.18.0

nmajor25 committed:
Update to Cypress v13

nmajor25 committed:
Refactor 'Alert Service => Displaying' test

Goal is to fix a test failure being caused by multiple cy.clock and clock.restore() in the same test

paulfalgout commented:
🥁 🧻

nmajor25 commented on test/integration/services/alert.js:
```diff
@@ -62,57 +62,63 @@ context('Alert Service', function() {
       .get('.alert-box')
       .find('.js-dismiss')
       .click()
-      .click()
```
There were two `.click()`s here... I removed it, but maybe that was needed?

nmajor25 replied:
Well I guess this is why there were 2 clicks there: https://coveralls.io/builds/62792353/source?filename=src%2Fjs%2Fviews%2Fglobals%2Falert-box%2Falert-box_views.js#L57.

Is for coverage purposes.

nmajor25 commented on test/integration/services/alert.js:
```diff
@@ -62,57 +62,63 @@ context('Alert Service', function() {
+
+    cy
+      .get('.alert-box')
+      .should('not.exist');
```
I added this. Just checks that the alert is no longer there after the undo button is clicked. Before we were only checking that the radio was sent/called.

nmajor25 committed:
Remove '.is-active' selector in 'Noncontext form => update patient field' test

paulfalgout reviewed (changes requested):
Alright.. I think we should keep everything but the `is-active` removal and upgrading to Cypress 13 for now.  the other stuff is either good, or won't hurt for now.

nmajor25 committed:
Revert "Remove '.is-active' selector in 'Noncontext form => update patient field' test"

This reverts commit 7e93c2d4f1b7310ca09af5dfbc95489dbdddbb90.

nmajor25 committed:
Downgrade back to Cypress v12

nmajor25 committed:
Update sass and autoprefixer npm packages

nmajor25 commented:
@paulfalgout I reverted the`.is-active` change and downgraded back to Cypress v12.

I also did a quick update of the `sass` and `autoprefixer` packages. Since those are new since the last time I did a `npm update` a few days ago.
